src/BenchmarkDotNet.Diagnostics.Windows/Tracing/NativeMemoryLogParser.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Diagnosers;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using Microsoft.Diagnostics.Tracing.Etlx;
+using Microsoft.Diagnostics.Tracing.Parsers.Kernel;
+using Address = System.UInt64;
+
+namespace BenchmarkDotNet.Diagnostics.Windows.Tracing
+{
+    public class NativeMemoryLogParser
+    {
+        private static readonly string LogSeparator = new string('-', 20);
+        private readonly string etlFilePath;
+        private readonly BenchmarkCase benchmarkCase;
+
+        private readonly ILogger logger;
+
+        public NativeMemoryLogParser(string etlFilePath, BenchmarkCase benchmarkCase, ILogger logger)
+        {
+            this.etlFilePath = etlFilePath;
+            this.benchmarkCase = benchmarkCase;
+            this.logger = logger;
+        }
+
+
+        //Code is inspired by https://github.com/Microsoft/perfview/blob/master/src/PerfView/PerfViewData.cs#L5719-L5944
+        public IEnumerable<Metric> Parse()
+        {
+            using (var traceLog = new TraceLog(TraceLog.CreateFromEventTraceLogFile(etlFilePath)))
+            {
+                var eventSource = traceLog.Events.GetSource();
+
+                var bdnEventsParser = new EngineEventLogParser(eventSource);
+
+                var start = false;
+                var isFirstActualStartEnd = false;
+
+                long totalOperation = 0;
+
+                bdnEventsParser.WorkloadActualStart += data =>
+                {
+                    if (!isFirstActualStartEnd)
+                    {
+                        start = true;
+                    }
+
+                    totalOperation = data.TotalOperations;
+                };
+                bdnEventsParser.WorkloadActualStop += data =>
+                {
+                    start = false;
+                    isFirstActualStartEnd = true;
+                };
+
+                var heapParser = new HeapTraceProviderTraceEventParser(eventSource);
+                // We index by heap address and then within the heap we remember the allocation stack
+                var heaps = new Dictionary<Address, Dictionary<Address, long>>();
+                Dictionary<Address, long> lastHeapAllocs = null;
+
+                Address lastHeapHandle = 0;
+
+                long nativeLeakSize = 0;
+                long totalAllocation = 0;
+
+                heapParser.HeapTraceAlloc += delegate(HeapAllocTraceData data)
+                {
+                    if (!start)
+                    {
+                        return;
+                    }
+
+                    var allocs = lastHeapAllocs;
+                    if (data.HeapHandle != lastHeapHandle)
+                        allocs = CreateHeapCache(data.HeapHandle, heaps, ref lastHeapAllocs, ref lastHeapHandle);
```

Comment:
nit: all other single line conditional blocks of code are wrapped with `{  }`, it would be nice to keep the consistency